Config UI: evcc.yaml editor

naltatis opened the pull request:
Integrate in-browse editing of evcc.yaml as a temporary solution until Config UI is done.

Depends on ~#9812 (layout, restart, dirty), #11732 (security)~

**Changes**

- ✏️ make `evcc.yaml` editable through config ui
- 🔐 only possible with authentication
- 🧪 removed experimental flag from "configuration" page. now only some features (vehicles, meters, ...) are hidden but not the complete page any more. yaml editor will be available as non-experimental
- 🔴🟢  edit > restart > test cycle on one page for faster iteration when making changes to yaml
- 🕸️ this feature will be removed once we've ported all configuration over to browser form based entry

**Todo**

- [x] secure API for loading/saving yaml (depends on #11732)
- [ ] no editor when evcc has no r/w permissions
- [x] workflow: save -> restart -> error/success
- [x] Decision: how does it relate to startup error page?
- [x] use breadcrumb for back navigaton (home / config / editor)
- [x] textarea fallback (no internet) https://github.com/imguolao/monaco-vue/issues/47
- [ ] add e2e tests


## Video

https://github.com/evcc-io/evcc/assets/152287/28fb4817-badb-4e12-b1aa-4ed852b6f1d2

## Screenshots

Light
![light](https://github.com/evcc-io/evcc/assets/152287/6a0c1160-b723-4835-937e-658917b41d14)

Dark
![dark](https://github.com/evcc-io/evcc/assets/152287/b4045b58-52b6-4496-a7c0-e4c19fb6f47b)

Dark Error
![dark error](https://github.com/evcc-io/evcc/assets/152287/bd5a1400-9c78-4c37-aa88-1fd9f3ef0ba5)

Dark Success
![dark success](https://github.com/evcc-io/evcc/assets/152287/d57f18e6-1e0f-4df6-910f-883277bd8646)

naltatis committed:
yaml editor protoype

StefanSchoof commented:
Very cool. How does this affect the asset size? Is there a need to do some chunking and lazy loading to load the editor only when needed?

naltatis commented:
@StefanSchoof the `vue-monaco-editor` library must be included as a Vue plugin. I had lazy-install on route change working for the config editor page. But this is quite messy in Vue. Now it's installed globally which is a lot easier. To good thing about this library is, thats its a very thin wrapper around Monaco. It only adds a few kb of js to the binary. The editor itself is quite mighty and heavy but the wrapper loads the acutal source from a CDN. See:

```
app.use(VueMonacoEditorPlugin, {
  paths: { vs: "https://cdn.jsdelivr.net/npm/monaco-editor@0.45.0/min/vs" },
});
```

I'm usually a big fan of self-contained software and not relying on third-party infrastructure, but in this case, I think the tradeoff is ok, since the user gets VSCode-grade syntax highlighting and folding. One thing I've just added to the todo list: make sure it also works in a no-internet situation: textarea-fallback.

naltatis committed:
merge

naltatis committed:
wip: r/w config file, http

naltatis committed:
offline fallback; layout

naltatis committed:
Merge branch 'master' into experiment/yaml_editor

naltatis committed:
restart, dirty, breadcrumb

naltatis committed:
editor improvements, config ui now completly experimental, translation, flow

naltatis committed:
error handling

naltatis committed:
userflow

naltatis committed:
fix integration tests

naltatis committed:
Merge branch 'master' into experiment/yaml_editor

andig committed:
Sunspec: Fix regression in model selection (#12274)

andig committed:
Add Tesla proxy client (#12174)

naltatis committed:
Custom select: add button semantics (#12279)

naltatis committed:
Visualization: hide values if not enough space (#12280)

naltatis committed:
Config UI: move generic meters to top (#12281)

naltatis committed:
Plan UI: debounce preview (#12284)

andig committed:
chore: better linter output

andig committed:
chore: use go 1.22 (#11007)

andig committed:
Tronity: improve status detection (#12296)

berndkrannich committed:
Fronius Solar API: add descriptions (#12310)

berndkrannich committed:
chore: don't set mask here as it is the default (#12313)

Signed-off-by: berndkrannich <github@krannich.net>

GrimmiMeloni committed:
Easee: always adjust DCC after resuming charge (#12258)

andig committed:
Tesla: unify implementations (BC) (#12277)

berndkrannich committed:
Fronius Solar API: explain limitations of active battery control (#12322)

Signed-off-by: berndkrannich <github@krannich.net>

andig committed:
Tesla: fix build

andig committed:
Powerwall: revert unification with tesla identity

andig committed:
Tesla: fix vehicle asleep detection

cschlipf committed:
Sofarsolar-g3: remove delay (#12341)

andig committed:
chore: fix logical order

andig committed:
chore: add test

naltatis committed:
Battery settings: improve buffer start

naltatis committed:
initial priority soc fix

andig committed:
Tesla: fix detecting vehicle asleep (#12369)

naltatis committed:
Publish as mDNS service

naltatis committed:
clean

andig committed:
Tesla: always expose the controller interface (#12391)

andig committed:
Add support for Amber Electric (AU) (#12381)

premultiply committed:
Bender: add Spelsberg and Cubos (#12405)

andig committed:
Modbus: fix int to float conversion for int setter

weblate committed:
Translations update from Hosted Weblate (#12299)

* Translated using Weblate (Dutch)

Currently translated at 100.0% (358 of 358 strings)

Co-authored-by: hyperbart <registeradressbart@gmail.com>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/nl/
Translation: evcc/evcc

* Translated using Weblate (Czech)

Currently translated at 36.8% (132 of 358 strings)

Co-authored-by: czkanec <facebook@janecka.info>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/cs/
Translation: evcc/evcc

* Translated using Weblate (Romanian)

Currently translated at 74.3% (266 of 358 strings)

Co-authored-by: infinus <azskylineGTR@gmail.com>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/ro/
Translation: evcc/evcc

* Tesla: fix vehicle asleep detection

* fix toml

---------

Co-authored-by: hyperbart <registeradressbart@gmail.com>
Co-authored-by: czkanec <facebook@janecka.info>
Co-authored-by: infinus <azskylineGTR@gmail.com>
Co-authored-by: andig <cpuidle@gmx.de>
Co-authored-by: premultiply <4681172+premultiply@users.noreply.github.com>

andig committed:
SolarEdge: re-add battery control (#11904)

nicok1 committed:
Offer rollback option for interactive upgrading  (#11771)

chrissooo committed:
Retry vehicle wake-up if necessary (#11707)

andig committed:
cli: allow setting vehicle current (if supported) (#12429)

andig committed:
chore: simplify how mqtt setters are defined (#12428)

andig committed:
chore: upgrade libraries

JanDragon committed:
chore: remove unused translation strings (#12426)

naltatis committed:
Deploy release and nightly doc templates separately (#12443)

* Deploy release and nightly doc templates seperately

* Update commit message

naltatis committed:
Smart grid charging per loadpoint (BC) (#12135)

deadrabbit87 committed:
Deye Hybrid 3p: add battery control  (#12415)

andig committed:
Sunpec: add split-phase inverters (#12446)

andig committed:
chore: add gofmt

andig committed:
Revert "chore: add gofmt"

This reverts commit 3e787b5a5f3380ad66fa49a1b0ffd2733816d44b.

andig committed:
chore: upgrade standard packages

andig committed:
Revert "chore: upgrade standard packages" due to https://github.com/golangci/golangci-lint/issues/4415

This reverts commit ad0c6d5eb1363e2ba554c3bbb30aac07798171f9.

naltatis committed:
fix: time-based price test

andig committed:
Huawei: add battery control (#12449)

andig committed:
Sofarsolar-g3: fix delay BC break

andig committed:
Ohmpilot: make id configurable (BC) (#12475)

andig committed:
Orno: set 8E1

andig committed:
Sunspec Inverter: fix battery (#12482)

andig committed:
RCT: fix retrying recoverable errors (#12476)

andig committed:
Refactor and simplify meter readings (#12479)

andig committed:
Huawei: fix register encoding

andig committed:
Huawei: specify modbus write types

andig committed:
Goodwe Wifi: fix retries not backed off (#12483)

andig committed:
Plugins: make const accept empty string as zero value

andig committed:
Polestar: re-login when token refresh fails (#12486)

naltatis committed:
Fix battery settings regressions (#12491)

naltatis committed:
App: prepare for embedding in native app (#12490)

andig committed:
Porsche: fix emobility api (#12500)

andig committed:
Tesla: clarify virtual key

naltatis committed:
UI: reduce height of main screen (#12507)

andig committed:
Retry pv meter reads on error (#12514)

andig committed:
Add my-PV ELWA (#12505)

naltatis committed:
Better offline detection (#12513)

ReneNulschDE committed:
Add Mercedes (#12403)

andig committed:
Revert "Modbus: fix int to float conversion for int setter"

This reverts commit 1d322c6330f2e60cd0bad884963c2a4136fbbd4c.

andig committed:
Revert "SolarEdge: re-add battery control (#11904)"

This reverts commit 8d2d95b431179206787edf513569fc0a05770d9f.

andig committed:
Tesla: document deprecation

premultiply committed:
Revert "chore: remove unused translation strings (#12426)"

This reverts commit 2e148e212338e9ec1f482c06855e55f1ae42cd39.

kscholty committed:
Add MG (#12003)

ReneNulschDE committed:
Mercedes: fix HTTP 502 when VIN is present and not equal FIN (#12535)

andig committed:
Reapply "Modbus: fix int to float conversion for int setter"

This reverts commit f9538eddedbdb24a5af8fe002b35a1a1ea858cd7.

andig committed:
Reapply "SolarEdge: re-add battery control (#11904)"

This reverts commit c930854d00d8926373250f1c7fbdd35bc4a50a9a.

naltatis committed:
Chore: Upgrade NPM dependencies (#12499)

* Chore: Upgrade NPM dependencies

* update

andig committed:
Ignore errors reading PV meter (#12564)

naltatis committed:
Fix fetching local http ressources on Chrome (#12546)

nicok1 committed:
chore: fix compatibility check (#12547)

ReneNulschDE committed:
Mercedes: update docs (#12569)

cschlipf commented:
Can I disable the editor (e.g. in the config) in case I do not want to expose this to anyone, who has access to EVCC?

In general the more editing capabilities the UI gets, the more we might need an authentication for a privileged user.

dependabot[bot] committed:
Bump golangci/golangci-lint-action from 3 to 4 (#12576)

Bumps [golangci/golangci-lint-action](https://github.com/golangci/golangci-lint-action) from 3 to 4.
- [Release notes](https://github.com/golangci/golangci-lint-action/releases)
- [Commits](https://github.com/golangci/golangci-lint-action/compare/v3...v4)

---
updated-dependencies:
- dependency-name: golangci/golangci-lint-action
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

cschlipf committed:
Sofar: document LSW-3 not supported (#12573)

Co-authored-by: Carsten <carsten.schlipf@microfocus.com>

andig committed:
RCT: fix retry

VolkerK62 committed:
Alfen: document ValidityTime setting (#12593)

andig committed:
Modbus: refactor write encoding (#12595)

andig committed:
Fix vehicle settings ignored when initialization fails (#12603)

TRON4R committed:
Config UI: fix typo in warning

andig committed:
chore: fix format

stumbaumr committed:
AC Elwa-E: fix temp (#12604)

motze92 committed:
Add PUN - Italienische Strompreise (#12262)

andig committed:
Mqtt: cleanup on start (#12621)

andig committed:
Tronity: fix app token refresh

andig committed:
chore: simplify

andig committed:
chore: minor

cmusik committed:
Huawei Smartlogger: add pv and battery support identical to SDongle & Power Sensor (#12628)

andig committed:
Ocpp: use setProfile=0 for suspend (#12459)

premultiply committed:
SAX Homespeicher: add battery control (#12462)

andig committed:
Planner: don't stop 100% plans (#12392)

andig committed:
chore: simplify setting limit (#12631)

andig committed:
Mqtt: fix error handling for tree cleanup

mabunixda committed:
Wattpilot: fix crash on ARM64 (#12648)

Signed-off-by: Martin Buchleitner <mabunixda@gmail.com>

StefanSchoof committed:
`evcc configure`: use non warning interval value (#12641)

naltatis committed:
Layout: no vertical scrolling (#12675)

naltatis committed:
Planner: faster loading (#12673)

andig committed:
chore: remove certificate pinning (before expiry) (#12670)

andig committed:
Revert "Publish as mDNS service"

This reverts commit 7db2d02b5d6c17ca614884a225393a654c1dfc61.

andig committed:
Revert "chore: remove certificate pinning (before expiry) (#12670)"

This reverts commit 041427928b9797b4efcebb7a2442c39c6c792e0c.

andig committed:
Reapply "chore: remove certificate pinning (before expiry) (#12670)"

This reverts commit 25fc95a7d45e18da993bf6e1dc726ade1815030c.

andig committed:
chore: fix missing GRPC config

andig committed:
AC Thor: add AC9 (#12721)

andig committed:
Mqtt: don't delete status topic

naltatis committed:
mDNS: read host from config (#12664)

* mDNS: readd host from config

* leaner

---------

Co-authored-by: andig <cpuidle@gmx.de>

andig committed:
Revert "chore: simplify"

This reverts commit 200c5b66c4ef6a3f6edc11b1157c7e86d013326e.

stumbaumr committed:
Add my-PV AC ELWA 2 (#12760)

As requested in https://github.com/evcc-io/evcc/pull/12604#issuecomment-1984038923

andig committed:
chore: simplify

andig committed:
Add Groupe E Vario Plus (CH) (#12767)

andig committed:
Bender: fix invalid readings (#12774)

andig committed:
Fix battery mode calculation across loadpoints (#12772)

naltatis committed:
Add battery color to visualization (#11733)

stumbaumr committed:
chore: rename ac-elwa-2 to ac-elwa-2.yaml (#12782)

JanDragon committed:
JLR: upgrade api (#12778)

VolkerK62 committed:
Add Plexlog (#12696)

naltatis committed:
MQTT: missing data (#12801)

naltatis committed:
Config UI: only reset form-data on manual template change (#12815)

naltatis committed:
Plan UI: confirm on unsaved changes (#12797)

naltatis committed:
Plan UI: improve mobile form layout (#12795)

naltatis committed:
Plan UI: relax plan not reachable detection (#12793)

naltatis committed:
Fix min-height on iOS homescreen (#12792)

andig committed:
Goodwe: fix grid energy

jpwenzel committed:
Elvi EVBox: disable getconfiguration (#12823)

andig committed:
Revert "Huawei Smartlogger: add pv and battery support identical to SDongle & Power Sensor (#12628)" (#12842)

This reverts commit fa5330b292c6da8c2a9c40e4c43710c748ebccfc.

andig committed:
chore: ensure valid config gets printed

andig committed:
chore: simplify and fix indents for postremove (#12849)

jpwenzel committed:
chore: fix Vetur spelling (#12855)

premultiply committed:
GoodWe-Hybrid: fix registers (#12854)

andig committed:
chore: simplify TWC3

andig committed:
chore: validate server name

andig committed:
Revert "Ocpp: use setProfile=0 for suspend (#12459)"

This reverts commit 25745b86b5d6ff3b51e4aba2f38757027de7f561.

jpwenzel committed:
Elvi: fix getConfiguration by adding delay (#12856)

andig committed:
Fix vehicle unable to override min current (#12866)

andig committed:
chore: remove out of band publish

MarkusGH committed:
Restore logic for chargers that disable themselves to be able to switch phases (#12847)

andig committed:
Setup: warn on illegal names

andig committed:
Groupe-E: remove charges and taxes

premultiply committed:
Add APsystems EZ1 inverters (#12874)

premultiply committed:
Goodwe-Hybrid: fix pv power

cmusik committed:
Huawei Smartlogger: add battery control similar to dongle configuration (#12877)

This is v2 of the configuration. The main difference is that it is
mostly backward compatible to the old smartlogger configuration. The
only change is that it is now using register 40521 (active power without
battery discharge) instead of 40525 (active power including battery
discharge).

The grid information comes from the powermeter which has usually the
unit id 11. The pv informations come directly from smartlogger (unit id
0, hardcoded). The battery information is fetched from the hybrid inverter.

A typical configuration should look like this:

```
meters:
- name: grid
  type: template
  template: huawei-smartlogger
  usage: grid
  id: 11
  host: host.fqdn
  port: 502

- name: pv
  type: template
  template: huawei-smartlogger
  usage: pv
  host: host.fqdn
  port: 502

- name: battery
  type: template
  template: huawei-smartlogger
  usage: battery
  id: 3
  host: host.fqdn
  port: 502
  capacity: 10
```

premultiply committed:
GoodWe-Hybrid: revert pv power to mppts

andig committed:
Ovms: add miles support (#12883)

andig committed:
VW/Audi: handle missing target soc

andig committed:
Add Weidmüller AC Smart (#12899)

andig committed:
chore: better error message for missing mask

andig committed:
chore: minor typo

alaingiller committed:
Smartfox: add energy for aux + update description (#12891)

andig committed:
Add Retryable feature to indicate vehicles that are temporarily offline (#12841)

andig committed:
chore: upgrade libraries

premultiply committed:
Mennekes: fix typo (#12932)

andig committed:
Revert "Mennekes: fix typo (#12932)"

This reverts commit c1a3d11f9af7d4d3887e585ff6228f33e7664126.

andig committed:
chore: simplify http method declaration

andig committed:
Tariffs: split naming by country

andig committed:
Add SmartEnergy tariff (#12971)

andig committed:
Modbus: fix writing bool values (#12972)

andiwist committed:
GoodWe Hybrid: add battery control (#12833)

andig committed:
chore: document battery control

naltatis committed:
Mobile UI: add active loadpoint indicator (#12973)

naltatis committed:
UI: fix header spacing (#12983)

andig committed:
Mennekes Compact: start heartbeat immediately

andig committed:
chore: collect tickers

andig committed:
OpenWB Pro: add rfid support (#13011)

andig committed:
Mennekes: fix type name typo (BC) (#12945)

Cytron1980 committed:
Solaredge Hybrid: clarify battery mode requirements  (#13022)

andig committed:
Mqtt: process messages in order (#13037)

andig committed:
chore: replace exp/slices

andig committed:
Remove battery/pv configured properties (#12498)

naltatis committed:
UI: reload on upgrade and startup error fix (#12835)

andig committed:
Webasto Next: clarify error status (#13048)

andig committed:
Vehicle: add maxcurrent setter (#13055)

thierolm committed:
Shelly: enable gen3 devices (#13057)

naltatis committed:
chore: upgrade npm deps (#13051)

weblate committed:
Translations update from Hosted Weblate (#12538)

* Translated using Weblate (Spanish)

Currently translated at 100.0% (367 of 367 strings)

Translated using Weblate (Spanish)

Currently translated at 100.0% (366 of 366 strings)

Translated using Weblate (Spanish)

Currently translated at 100.0% (365 of 365 strings)

Translated using Weblate (Spanish)

Currently translated at 100.0% (357 of 357 strings)

Co-authored-by: gallegonovato <fran-carro@hotmail.es>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/es/
Translation: evcc/evcc

* Translated using Weblate (Slovenian)

Currently translated at 100.0% (357 of 357 strings)

Co-authored-by: Žiga Deisinger <ziga@deisinger.si>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/sl/
Translation: evcc/evcc

* Translated using Weblate (Romanian)

Currently translated at 100.0% (357 of 357 strings)

Co-authored-by: Bogdan Gheorghe <bogdan.gheorghe@gmail.com>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/ro/
Translation: evcc/evcc

* Translated using Weblate (Danish)

Currently translated at 100.0% (365 of 365 strings)

Translated using Weblate (Danish)

Currently translated at 100.0% (357 of 357 strings)

Co-authored-by: amtssp <amtssp@gmail.com>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/da/
Translation: evcc/evcc

* Translated using Weblate (Ukrainian)

Currently translated at 91.0% (334 of 367 strings)

Translated using Weblate (Ukrainian)

Currently translated at 89.1% (327 of 367 strings)

Translated using Weblate (Polish)

Currently translated at 100.0% (367 of 367 strings)

Translated using Weblate (Lithuanian)

Currently translated at 100.0% (367 of 367 strings)

Translated using Weblate (Ukrainian)

Currently translated at 87.4% (320 of 366 strings)

Translated using Weblate (Ukrainian)

Currently translated at 84.4% (309 of 366 strings)

Translated using Weblate (Ukrainian)

Currently translated at 82.7% (303 of 366 strings)

Translated using Weblate (Polish)

Currently translated at 100.0% (366 of 366 strings)

Translated using Weblate (Lithuanian)

Currently translated at 100.0% (366 of 366 strings)

Translated using Weblate (Danish)

Currently translated at 100.0% (365 of 365 strings)

Translated using Weblate (Swedish)

Currently translated at 95.3% (348 of 365 strings)

Translated using Weblate (Ukrainian)

Currently translated at 80.5% (294 of 365 strings)

Translated using Weblate (Spanish)

Currently translated at 100.0% (365 of 365 strings)

Translated using Weblate (German)

Currently translated at 98.9% (361 of 365 strings)

Translated using Weblate (Lithuanian)

Currently translated at 100.0% (365 of 365 strings)

Translated using Weblate (Polish)

Currently translated at 100.0% (365 of 365 strings)

Translated using Weblate (Lithuanian)

Currently translated at 100.0% (365 of 365 strings)

Co-authored-by: RTTTC <94727758+RTTTC@users.noreply.github.com>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/da/
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/de/
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/es/
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/lt/
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/pl/
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/sv/
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/uk/
Translation: evcc/evcc

* Translated using Weblate (French)

Currently translated at 100.0% (367 of 367 strings)

Translated using Weblate (French)

Currently translated at 100.0% (366 of 366 strings)

Translated using Weblate (French)

Currently translated at 100.0% (365 of 365 strings)

Co-authored-by: FraBoCH <fraboch@proton.me>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/fr/
Translation: evcc/evcc

* Translated using Weblate (Swedish)

Currently translated at 97.8% (358 of 366 strings)

Translated using Weblate (Swedish)

Currently translated at 97.8% (357 of 365 strings)

Co-authored-by: Jonas Gate <tidy.koala8317@brot.se>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/sv/
Translation: evcc/evcc

* Translated using Weblate (Czech)

Currently translated at 100.0% (365 of 365 strings)

Translated using Weblate (Czech)

Currently translated at 95.3% (348 of 365 strings)

Translated using Weblate (Czech)

Currently translated at 86.5% (316 of 365 strings)

Co-authored-by: petosak <petosak@gmail.com>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/cs/
Translation: evcc/evcc

* Translated using Weblate (Danish)

Currently translated at 100.0% (367 of 367 strings)

Translated using Weblate (Danish)

Currently translated at 100.0% (366 of 366 strings)

Translated using Weblate (Danish)

Currently translated at 100.0% (365 of 365 strings)

Co-authored-by: amtssp <amtssp@gmail.com>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/da/
Translation: evcc/evcc

* Translated using Weblate (German)

Currently translated at 99.4% (365 of 367 strings)

Co-authored-by: Gregor <it.is+weblate@posteo.de>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/de/
Translation: evcc/evcc

* Translated using Weblate (Czech)

Currently translated at 100.0% (367 of 367 strings)

Co-authored-by: czkanec <facebook@janecka.info>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/cs/
Translation: evcc/evcc

* Translated using Weblate (Dutch)

Currently translated at 99.4% (365 of 367 strings)

Co-authored-by: hyperbart <registeradressbart@gmail.com>
Translate-URL: https://hosted.weblate.org/projects/evcc/evcc/nl/
Translation: evcc/evcc

* fix toml

---------

Co-authored-by: gallegonovato <fran-carro@hotmail.es>
Co-authored-by: Žiga Deisinger <ziga@deisinger.si>
Co-authored-by: Bogdan Gheorghe <bogdan.gheorghe@gmail.com>
Co-authored-by: amtssp <amtssp@gmail.com>
Co-authored-by: RTTTC <94727758+RTTTC@users.noreply.github.com>
Co-authored-by: FraBoCH <fraboch@proton.me>
Co-authored-by: Jonas Gate <tidy.koala8317@brot.se>
Co-authored-by: petosak <petosak@gmail.com>
Co-authored-by: Gregor <it.is+weblate@posteo.de>
Co-authored-by: czkanec <facebook@janecka.info>
Co-authored-by: hyperbart <registeradressbart@gmail.com>
Co-authored-by: premultiply <4681172+premultiply@users.noreply.github.com>

JanDragon committed:
JLR: fix api (#13068)

andig committed:
Huawei: fix charge from grid register

pmknowles committed:
Ngeso: correct Co2 configuration (#13085)

naltatis committed:
UI: dark menubar color on modal (#13102)

andig committed:
chore: allow overriding image name

andig committed:
Loadpoint: log specific charger errors (#13097)

berndkrannich committed:
chore: fix telemetry details link (#13115)

Signed-off-by: berndkrannich <github@krannich.net>

andig committed:
chore: remove dead code

andig committed:
chore: upgrade libraries

andig committed:
chore: allow overriding image name (cont.)

andig committed:
chore: add TESLA_CLIENT_ID to local docker builds

andig committed:
Custom vehicle: add charge controller (#13119)

andiwist committed:
GoodWe: fix battery control (#13125)

andig committed:
Custom vehicle: add limitsoc (#13100)

andig committed:
Tesla: validate vehicle current (#13099)

andig committed:
chore: upgrade alpine image (#13128)

andig committed:
chore: cleanup

andig committed:
Custom vehicle: add getMaxCurrent (#13138)

kscholty committed:
Saic: improve response handling (#12993)

Adminius committed:
Add TeslaLogger vehicle (#13046)

naltatis committed:
Merge branch 'master' into experiment/yaml_editor

naltatis committed:
build fix

naltatis committed:
lint fix

naltatis committed:
lint fix

naltatis committed:
Merge branch 'master' into experiment/yaml_editor

naltatis committed:
Merge branch 'master' into experiment/yaml_editor

naltatis committed:
Merge branch 'master' into experiment/yaml_editor

naltatis committed:
auth reenable

naltatis committed:
combine yaml editor with auth

naltatis committed:
layout; i18n
